data-prepper-api/src/test/java/org/opensearch/dataprepper/typeconverter/LongConverterTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ package org.opensearch.dataprepper.typeconverter;
+
+ import org.junit.jupiter.api.Test;
+ 
+ import static org.hamcrest.CoreMatchers.equalTo;
+ import static org.hamcrest.MatcherAssert.assertThat;
+ import static org.junit.jupiter.api.Assertions.assertThrows;
+ 
+ public class LongConverterTests {
+     @Test
+     void testStringToLongConversion() {
+         LongConverter converter = new LongConverter();
+         final String stringConstant = "1234";
+         assertThat(converter.convert(stringConstant), equalTo(Long.parseLong(stringConstant)));
+     }
+     @Test
+     void testfloatToLongConversion() {
+         LongConverter converter = new LongConverter();
+         final Float floatConstant = (float)1234.56789;
+         assertThat(converter.convert(floatConstant), equalTo((long)(float)floatConstant));
+     }
+     @Test
+     void testDoubleToLongConversion() {
+        LongConverter converter = new LongConverter();
+        final Double doubleConstant = (double)12345678.12345678;
+        assertThat(converter.convert(doubleConstant), equalTo((long)(double)doubleConstant));
+     }
+     @Test
```

Comment:
These test cases look good. One thing you could do to make it easier to add more tests would be to use @ParameterizedTest with an Arguments source. An example of that can be found here (https://github.com/opensearch-project/data-prepper/blob/8cb5a8463719af83dfe9b88fcd614f191ea16767/data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluator_ArithmeticIT.java#L113)

Reply:
Added the `ParameterizedTest` for all the test except for `testInvalidStringConversion` as had to create a `MethodSource`.